server/src/main/java/io/druid/server/metrics/MetricsModule.java

```diff
@@ -107,39 +106,37 @@ public MonitorScheduler getMonitorScheduler(
 
   @Provides
   @ManageLifecycle
-  public JvmMonitor getJvmMonitor(Properties props)
+  public JvmMonitor getJvmMonitor(
+      DataSourceTaskIdHolder dataSourceTaskIdHolder
```

Comment:
`DataSourceTaskIdHolder` is bound in `MetricsModule` which is loaded as part of the ServerRunnable list of modules to load